cltk/prosody/old_norse/verse.py

```diff
@@ -0,0 +1,214 @@
+"""Verse structures of Old Norse poetry"""
+
+from math import floor
+from cltk.phonology.old_norse.transcription import Transcriber,  old_norse_rules, IPA_class, DIPHTHONGS_IPA_class, \
+    DIPHTHONGS_IPA
+from cltk.phonology.syllabify import Syllabifier
+from cltk.tokenize.word import tokenize_old_norse_words
+from cltk.utils.cltk_logger import logger
+
+__author__ = ["Clément Besnier <clemsciences@aol.com>", ]
+
+
+class VerseManager:
+    """
+    * Fornyrðislag
+    * Ljóðaháttr
+    """
+    @staticmethod
+    def is_fornyrdhislag(text: str):
+        """
+        Basic check
+        >>> text1 = "Hljóðs bið ek allar\\nhelgar kindir,\\nmeiri ok minni\\nmögu Heimdallar;\\nviltu at ek, Valföðr,\\nvel fyr telja\\nforn spjöll fira,\\nþau er fremst of man."
+        >>> text2 = "Deyr fé,\\ndeyja frændr,\\ndeyr sjalfr it sama,\\nek veit einn,\\nat aldrei deyr:\\ndómr um dauðan hvern."
+        >>> VerseManager.is_fornyrdhislag(text1)
+        True
+        >>> VerseManager.is_fornyrdhislag(text2)
+        False
+
+        :param text:
+        :return:
+        """
+        l = [line for line in text.split("\n") if line != ""]
+        return len(l) == 8
+
+    @staticmethod
+    def is_ljoodhhaattr(text: str):
+        """
+        Basic check*
+
+        >>> text1 = "Hljóðs bið ek allar\\nhelgar kindir,\\nmeiri ok minni\\nmögu Heimdallar;\\nviltu at ek, Valföðr,\\nvel fyr telja\\nforn spjöll fira,\\nþau er fremst of man."
+        >>> text2 = "Deyr fé,\\ndeyja frændr,\\ndeyr sjalfr it sama,\\nek veit einn,\\nat aldrei deyr:\\ndómr um dauðan hvern."
+        >>> VerseManager.is_ljoodhhaattr(text1)
+        False
+        >>> VerseManager.is_ljoodhhaattr(text2)
+        True
+
+        :param text:
+        :return:
+        """
+        l = [line for line in text.split("\n") if line != ""]
+        return len(l) == 6
+
+
+class Verse:
+    def __init__(self):
+        self.short_lines = []
+        self.long_lines = []
+        self.syllabified_text = []
+        self.transcribed_text = []
+
+    def from_short_lines_text(self, text: str):
+        pass
+
+    def syllabify(self):
+        """
+
+        """
+        if len(self.long_lines) == 0:
+            logger.error("No text was imported")
+            self.syllabified_text = []
+        else:
+            syllabifier = Syllabifier(language="old_norse", break_geminants=True)
+            syllabified_text = []
+            for i, line in enumerate(self.long_lines):
+                syllabified_text.append([])
+                for j, viisuordh in enumerate(line):
+                    syllabified_text[i].append([])
+                    words = []
+                    for word in tokenize_old_norse_words(viisuordh):
```

Comment:
a nitpick, but why not replace it with regex? You would also get an extra boost in speed.